Library/Homebrew/dev-cmd/test-bot.rb

```diff
@@ -920,7 +920,6 @@ def sanitize_ARGV_and_ENV
     end
 
     ENV["HOMEBREW_DEVELOPER"] = "1"
-    ENV["HOMEBREW_NO_AUTO_UPDATE"] = "1"
```

Comment:
> brew install --only-dependencies --build-bottle --verbose mesos

Because the `test-bot` was run through `brew.sh` which has already exported `HOMEBREW_NO_AUTO_UPDATE`. It'll basically handle all cases where a `brew` invocation ends up calling other processes. This also handles the case for e.g. `brew bundle` and other external commands that may call `brew install` in quick succession.